### ISSUE TITLE
Internal error,   ("Invalid Package.Name.t", { s = "" })

### ISSUE THREAD

**Issue by superherointj** [ocaml/dune#3584]
## Expected Behavior
Shouldn't crash? I'm only reporting because Dune said I should report it upstream. Maybe it is relevant?

## Actual Behavior
"Internal error, please report upstream including the contents of _build/log."
Description:
  ("Invalid Package.Name.t", { s = "" })
Raised at file "src/stdune/code_error.ml", line 9, characters 30-62

```
I must not crash.  
Uncertainty is the mind-killer. 
Exceptions are the little-death that brings total obliteration.  
I will fully express my cases. 
Execution will pass over me and through me.  
And when it has gone past, I will unwind the stack along its path.  
Where the cases are handled there will be nothing.  
Only I will remain.
```

## Reproduction
Sample project:
https://github.com/superherointj/rely-mock-demo/

Just run "dune build -p demo1, demo1tests".

Or 'make' for running inside Docker.

## Specifications

- Version of `dune` (output of `dune --version`): 2.6.0
- Version of `ocaml` (output of `ocamlc --version`): 4.10.0
- Operating system (distribution and version): Arch Linux 5.7.6-arch1-1

## Additional information 

- Logs are available at:
https://github.com/superherointj/rely-mock-demo/blob/master/logs/Docker.logs

**Comment by superherointj** [comment]
I was able to isolate error to having space between package names like:
``
dune build -p demo1, demo1tests
``

By removing the space the error goes away.

**Comment by jeremiedimino** [comment]
Such a crash would still be a bug BTW, Dune shouldn't crash badly like this on user errors. Though this one seems to have been fixed already.

**Comment by superherointj** [comment]
The demo GIT repo didn't have the error because I have removed the space as a test. I've reversed the change to check if it is fixed. And it is not. (A git pull is required) I've just confirmed that Dune 2.6.0 still crashes. Was this fixed for in a future version?

**Comment by jeremiedimino** [comment]
I get this with master:

```
$ dune build -p x,
dune: option `-p': Invalid package name: ""
Usage: dune build [OPTION]... [TARGET]...
Try `dune build --help' or `dune --help' for more information.
```

which seems like the right behaviour to me. i.e. you get a nice error message rather than a backtrace. Dune failing in this case is expected.

**Comment by superherointj** [comment]
Yes, it is fixed in master (2.5.1-303-g47649a3d), just not in dune 2.60. Closed.
